examples/src/bin/composite_template.rs

```diff
@@ -0,0 +1,106 @@
+//! # Composite Template Example
+//!
+//! This sample demonstrates how to create a widget using GTK's composite templates.
+
+#[macro_use]
+extern crate glib;
+#[macro_use]
+extern crate gtk;
+
+use gio::{prelude::ApplicationExtManual, ApplicationExt};
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::{Cast, StaticType};
+use gtk::subclass::prelude::*;
+use gtk::subclass::widget::*;
+use gtk::WidgetExt;
+
+/// The private struct, which can hold widgets and other data.
+#[derive(Debug, CompositeTemplate)]
+pub struct ExApplicationWindowPriv {
+    // The #[template_child] attribute tells the CompositeTemplate macro
+    // that a field is meant to be a child within the template.
+    #[template_child(id = "headerbar")]
+    headerbar: TemplateChild<gtk::HeaderBar>,
+    #[template_child(id = "label")]
+    label: TemplateChild<gtk::Label>,
+    #[template_child(id = "subtitle_label")]
+    subtitle: TemplateChild<gtk::Label>,
+}
+
+impl ObjectSubclass for ExApplicationWindowPriv {
+    const NAME: &'static str = "ExApplicationWindow";
+    type Type = ExApplicationWindow;
+    type ParentType = gtk::ApplicationWindow;
+    type Instance = subclass::simple::InstanceStruct<Self>;
+    type Class = subclass::simple::ClassStruct<Self>;
+
+    glib_object_subclass!();
+
+    fn new() -> Self {
+        Self {
+            headerbar: TemplateChild::default(),
+            label: TemplateChild::default(),
+            subtitle: TemplateChild::default(),
+        }
+    }
+
+    // Within class_init() you must set the template
+    // and bind it's children. The CompositeTemplate
+    // derive macro provides a convenience function
+    // bind_template_children() to bind all children
+    // at once.
+    fn class_init(klass: &mut Self::Class) {
+        let template = include_bytes!("composite_template.ui");
+        Self::set_template(klass, template);
+        Self::bind_template_children(klass);
+    }
+}
+
+impl ObjectImpl for ExApplicationWindowPriv {}
+impl WidgetImpl for ExApplicationWindowPriv {}
+impl WindowImpl for ExApplicationWindowPriv {}
+impl ApplicationWindowImpl for ExApplicationWindowPriv {}
+
+glib_wrapper! {
+    pub struct ExApplicationWindow(ObjectSubclass<ExApplicationWindowPriv>)
+        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow, @implements gio::ActionMap, gio::ActionGroup;
+}
+
+impl ExApplicationWindow {
+    pub fn new<P: glib::IsA<gtk::Application> + glib::value::ToValue>(app: &P) -> Self {
```

Comment:
```suggestion
    pub fn new<P: glib::IsA<gtk::Application>>(app: &P) -> Self {
```

After https://github.com/gtk-rs/gtk-rs/pull/156